capabilityreceiver/cmd/migratevalidations/main_integration_test.go

```diff
@@ -0,0 +1,272 @@
+	defer valRows.Close()
+
+	for valRows.Next() {
+		var valCount int
+		err = valRows.Scan(&valCount)
+		th.Assert(t, valCount == 5, fmt.Sprintf("there should be 5 entries in the validations table with id %d, instead there are %d", valID, valCount))
+		count++
+	}
+
+	// Add another instance of the second URL
+	thirdTime := time.Now().UTC().Round(time.Microsecond)
+	_, err = store.DB.ExecContext(ctx, addFHIREndpointInfoStatement, url2, "U", capStat2, tlsVersion, pq.Array(mimeTypes), metadataID2, thirdTime)
+	th.Assert(t, err == nil, fmt.Sprintf("Error when adding to the database third time %s", err))
+
+	// Make sure all instances of that are updated
+	resultCh2 := make(chan Result)
+
+	// Check that data only updates the second URL
+	defaultArgs2 := make(map[string]interface{})
+	defaultArgs2["historyArgs"] = historyArgs{
+		fhirURL:   url2,
+		store:     store,
+		result:    resultCh2,
+		isHistory: true,
+	}
+
+	go addToValidationTable(ctx, &defaultArgs2)
+	for res := range resultCh2 {
+		th.Assert(t, res.URL == url2, fmt.Sprintf("Returned result URL is not equal to %s, is instead %s", url1, res.URL))
+		close(resultCh2)
+	}
+	historyRows, err = store.DB.QueryContext(ctx, getFHIREndpointInfoStatement, url2)
+	th.Assert(t, err == nil, fmt.Sprintf("error getting data from fhir_endpoints_info: %s", err))
+	// Check that both entries with url2 have been updated and that they don't have the same validation result ID
+	count = 0
+	var firstValID int
+	for historyRows.Next() {
+		th.Assert(t, count < 2, fmt.Sprintf("should be two items in the database for this URL"))
+		var receivedTime time.Time
+		var currentValID int
+		err = historyRows.Scan(&receivedTime, &currentValID)
+		th.Assert(t, err == nil, fmt.Sprintf("Error while scanning the rows of the history table for URL %s. Error: %s", url1, err))
+		th.Assert(t, currentValID != 0, fmt.Sprintf("The validation ID was not set for this history table entry %s", url1))
+		if count == 1 {
+			th.Assert(t, currentValID != firstValID, fmt.Sprintf("The second ID should not be equal to the first, %d, %d", currentValID, firstValID))
+		} else if count == 0 {
+			firstValID = currentValID
+		}
+		count++
+	}
+	th.Assert(t, count == 2, "should be two items in the database, instead is 1")
+
+	// Just check the first validation ID to make sure it's in validation_results
+	valResRows, err = store.DB.QueryContext(ctx, getValidationResultStatement, firstValID)
+	th.Assert(t, err == nil, fmt.Sprintf("error getting data from validation_results: %s", err))
+	defer valResRows.Close()
+
+	for valResRows.Next() {
+		var valResCount int
+		err = valResRows.Scan(&valResCount)
+		th.Assert(t, valResCount == 1, fmt.Sprintf("for URL %s, there should be one row with id %d", url1, firstValID))
+		count++
+	}
+
+	// Then check that it's 5 validation entries were added to the validation table
+	valRows, err = store.DB.QueryContext(ctx, getValidationStatement, valID)
```

Comment:
Here again you could use probably QueryRowContext to directly scan the count rather than having to call valRows.next(), since this query should only return one row anyways